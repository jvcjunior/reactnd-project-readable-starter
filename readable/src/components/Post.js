import React from 'react';
import {Card, CardActions, CardHeader, CardMedia, CardTitle, CardText} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import IconButton from 'material-ui/IconButton';
import FontIcon from 'material-ui/FontIcon';

const Post = ({post}) => {
    return (
        <Card key={post.id}>
            <CardHeader
                title={post.title}
                subtitle={post.author}
            />
            <CardText>
                <div>{post.body}</div>
                <div>Score: {post.voteScore}</div>
                <div>Comments: {post.commentCount}</div>
            </CardText>
            <CardActions>
                <FlatButton
                    icon={<FontIcon className="material-icons">mood</FontIcon>} />
                <FlatButton
                    icon={<FontIcon className="material-icons">mood_bad</FontIcon>} />
                <FlatButton label="Edit" />
                <FlatButton label="Delete" />
            </CardActions>
        </Card>
    );
}

export default Post;
