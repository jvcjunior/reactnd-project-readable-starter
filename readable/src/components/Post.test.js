import React from 'react';
import { shallow } from 'enzyme';
import Post from './Post';

describe('Post Component', () => {
    const post = shallow(<Post />);

    it('renders Post component correctly', () => {
        expect(post).toMatchSnapshot();
    });
});