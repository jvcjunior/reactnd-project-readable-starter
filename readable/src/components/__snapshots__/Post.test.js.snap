// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Component renders Post component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Post />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Titulo
</h1>,
        <h2>
          Autor
</h2>,
        <p>
          Número de comentários
</p>,
        <p>
          Pontuação atual
</p>,
        <p>
          Mecanismo de voto
</p>,
        <button>
          Editar
</button>,
        <button>
          Excluir
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Titulo",
        },
        "ref": null,
        "rendered": "Titulo",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Autor",
        },
        "ref": null,
        "rendered": "Autor",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Número de comentários",
        },
        "ref": null,
        "rendered": "Número de comentários",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Pontuação atual",
        },
        "ref": null,
        "rendered": "Pontuação atual",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mecanismo de voto",
        },
        "ref": null,
        "rendered": "Mecanismo de voto",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Editar",
        },
        "ref": null,
        "rendered": "Editar",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Excluir",
        },
        "ref": null,
        "rendered": "Excluir",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Titulo
</h1>,
          <h2>
            Autor
</h2>,
          <p>
            Número de comentários
</p>,
          <p>
            Pontuação atual
</p>,
          <p>
            Mecanismo de voto
</p>,
          <button>
            Editar
</button>,
          <button>
            Excluir
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Titulo",
          },
          "ref": null,
          "rendered": "Titulo",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Autor",
          },
          "ref": null,
          "rendered": "Autor",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Número de comentários",
          },
          "ref": null,
          "rendered": "Número de comentários",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pontuação atual",
          },
          "ref": null,
          "rendered": "Pontuação atual",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mecanismo de voto",
          },
          "ref": null,
          "rendered": "Mecanismo de voto",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Editar",
          },
          "ref": null,
          "rendered": "Editar",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Excluir",
          },
          "ref": null,
          "rendered": "Excluir",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
